api/core.py

```diff
@@ -19,6 +19,64 @@
 
 log = config.log
 
+def _filter_check(property_filter, property_values):
+    minus = set(property_filter.get('-', []))
+    plus = set(property_filter.get('+', []))
+    if not minus.isdisjoint(property_values):
+        return False
+    if plus and plus.isdisjoint(property_values):
+        return False
+    return True
+
+
+def _append_targets(targets, container, prefix, total_size, total_cnt, optional, data_path, filters):
+    for f in container.get('files', []):
+        if filters:
+            # Here we use filters in the payload to exclude/include files.
+            # To pass a single filter, each of its conditions should be satisfied.
+            # If a file pass at least one filter, it is included in the targets.
+            # For example:
+            #
+            # download_payload = {
+            #     'optional': True,
+            #     'nodes': [{'level':'project', '_id':project_id}],
+            #     'filters':[{
+            #         'tags':{'+':['incomplete']}
+            #     },
+            #     {
+            #         'types':{'-':['dicom']}
+            #     }]
+            # }
+            # will download files with tag 'incomplete' OR type different from 'dicom'
+            #
+            # download_payload = {
+            #     'optional': True,
+            #     'nodes': [{'level':'project', '_id':project_id}],
+            #     'filters':[{
+            #         'tags':{'+':['incomplete']},
+            #         'types':{'+':['dicom']}
+            #     }]
+            # }
+            # will download only files with tag 'incomplete' AND type different from 'dicom'
```

Comment:
Wouldn't this long comment be better placed into the docstring of the download method?